Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -155,6 +155,12 @@ public override void Analyze(AnalyzeContext context)
             {
                 try
                 {
+                    var fileInfo = new FileInfo(filePath);
```

Comment:
>FileInfo [](http://example.com/codeflow?start=39&length=8)

Always be careful when introducing file operations in code! You can't mock properly by calling this code directly. 

You should be operating against a FileSystem instance. You will need to create a helper in the default IFileSystem implementation that accepts a file path and computes the length (i.e., convert this pattern to a simple method)